### ISSUE TITLE
test(checks): Add automated bundle testing in pipeline

### ISSUE THREAD

**Issue by simar7** [aquasecurity/trivy#7031]
Today when a PR is submitted in the checks repo,  we have some basic testing that takes place as described in [this](https://github.com/aquasecurity/trivy-checks/blob/main/.github/workflows/test-bundle.yml) workload. However, it offers no integration testing with Trivy itself.

For this to take place, we can do the following:

1. Create a test environment with certain misconfigurations within the trivy-checks repo
2. Create a bundle with the changes submitted in the PR within a GitHub Action pipeline
3. Load this bundle into the latest release of Trivy, will be implemented via https://github.com/aquasecurity/trivy/issues/7029
4. Make assertions on the checks themselves but also warrant that the bundle is loaded successfully. 

By adding this automated layer of testing two problems will be tackled:

1. Avoiding surprise breakages when upgrading trivy-checks usage within Trivy.
2. Enabling automated scheduled releases of trivy-checks bundles without operator intervention. More details here: https://github.com/aquasecurity/trivy/issues/7030

**Comment by nikpivkin** [comment]
CI in trivy-checks is already doing integration testing of the checks bundle with the latest version of Trivy and seems to meet all the requirements described here.  https://github.com/aquasecurity/trivy-checks/blob/main/scripts/verify-bundle.go

**Comment by simar7** [comment]
> CI in trivy-checks is already doing integration testing of the checks bundle with the latest version of Trivy and seems to meet all the requirements described here. https://github.com/aquasecurity/trivy-checks/blob/main/scripts/verify-bundle.go

Yes but there's currently no functional verification. We also lack testing against multiple versions of Trivy as we only test `latest` right now.

**Comment by simar7** [comment]
Fixed via https://github.com/aquasecurity/trivy-checks/commit/532cf7689eb0a4b91f3c802470f2244058506b46
